website/src/components/Submission/FileUpload/ColumnMappingModal.tsx

```diff
@@ -0,0 +1,291 @@
+                    'Loading ...'
+                ) : (
+                    <div className='space-y-4'>
+                        <table>
+                            <thead>
+                                <tr>
+                                    <th className='py-2 sm:min-w-56'>Column in your file</th>
+                                    <th style={minWidthStyle}>Submission column</th>
+                                </tr>
+                            </thead>
+                            <tbody>
+                                {currentMapping.entries().map(([sourceCol, targetCol]) => (
+                                    <ColumnSelectorRow
+                                        key={sourceCol}
+                                        selectingFor={sourceCol}
+                                        selectedOption={targetCol}
+                                        options={groupedInputFields}
+                                        usedOptions={currentMapping.usedColumns()}
+                                        setColumnMapping={setCurrentMapping}
+                                    />
+                                ))}
+                            </tbody>
+                        </table>
+                        <div className='min-h-6 text-sm'>
+                            {missingFields.length > 0 && 'All required fields need to be set to apply this mapping.'}
+                        </div>
+                        <div className='flex flex-row gap-2 justify-end'>
+                            {columnMapping !== null && (
+                                <>
+                                    <button
+                                        className='btn bg-white text-red-800 border-red-800'
+                                        onClick={handleDiscard}
+                                    >
+                                        Discard Mapping
+                                    </button>
+                                    <div className='flex-1' />
+                                </>
+                            )}
+                            <button className='btn' onClick={closeDialog}>
+                                Cancel
+                            </button>
+                            <button
+                                className='btn loculusColor text-white'
+                                onClick={handleSubmit}
+                                disabled={!submittable}
+                            >
+                                {saveButtonText}
+                            </button>
+                        </div>
+                    </div>
+                )}
+            </BaseDialog>
+        </>
+    );
+};
+
+async function extractColumns(tsvFile: ProcessedFile): Promise<Result<string[], Error>> {
+    let text;
+    try {
+        // there is potential to optimize: don't read the whole file, just the header (read in chunks)
```

Comment:
is this an open TODO or can it be removed?

Reply:
I didn't write TODO because I didn't intend _to do_ it (😄)  but thought it'd still be a good note to have there. But maybe I just remove it, I guess from the code it's obvious that this improvement could be made; and the whole file will still be read later in the process anyways, so we wouldn't even gain that much. i'll remove it.

Reply:
removed: https://github.com/loculus-project/loculus/pull/3478/commits/49108a0f9f438b9b8790b0b6735c4e7b2a574a88